### PR TITLE
🔨 Refactor(/app/pages/Dashboard/index): replace Cerebral with 🧠 Overmind

### PR DESCRIPTION
**Pull request by armujahid**
## What kind of change does this PR introduce?

Refactor `/app/pages/Dashboard/index.js` #2621 @Saeris @christianalfoni

## What is the current behavior?

It was using cerebral for state management

## What is the new behavior?

It uses Overmind for state management with help of useOvermind hook

## What steps did you take to test this? This is required before we can merge.

1. yarn lint
2. yarn typecheck (Currently failing. @Saeris @christianalfoni kindly guide how to fix this type error)
image

3. yarn test

## Checklist

- [ ] Documentation (N/A)
- [x] Testing 
- [x] Ready to be merged
      
- [x] Added myself to contributors table (Already requested in https://github.com/codesandbox/codesandbox-client/pull/2832)

### CODE REVIEW

File: packages/app/src/app/pages/Dashboard/index.js [hunk 1]
```diff
@@ -22,89 +22,85 @@ import {
...
+    state: { hasLogIn },
+    actions: { dashboard },
+  } = useOvermind();
 
-  componentWillUnmount() {
-    // Reset store so new visits get fresh data
-    client.resetStore();
-  }
+  useEffect(() => {
+    dashboard.dashboardMounted();
```
Comment (armujahid) [inline packages/app/src/app/pages/Dashboard/index.js]: How to fix this type error ? 🤔
image

Comment (armujahid) [reply packages/app/src/app/pages/Dashboard/index.js]: All CI checks (including typecheck) are green so It seems that this typecheck error was my environment specific.

Comment (SaraVieira) [reply packages/app/src/app/pages/Dashboard/index.js]: May be an old thing that already got fixed

Comment (armujahid) [reply packages/app/src/app/pages/Dashboard/index.js]: CI typecheck is now failing because of this error after renaming the file to tsx.

**Comment by armujahid** [comment]
@SaraVieira done

File: packages/app/src/app/pages/Dashboard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  if (!hasLogIn) {
+    DashboardContent = (
+      <Container>
+        <Centered>
+          <LoggedInContainer>
+            <OffsettedLogo />
+            <LoggedInTitle>Sign in to CodeSandbox</LoggedInTitle>
+
+            <SignInButton big style={{ fontSize: '1rem' }} />
```
Comment (armujahid) [inline packages/app/src/app/pages/Dashboard/index.tsx]: Second typecheck error was already present in unmodified file
image

Comment (armujahid) [reply packages/app/src/app/pages/Dashboard/index.tsx]: I can fix this by adding `big?: boolean;` in https://github.com/codesandbox/codesandbox-client/blob/a62cec439d1f80d991ad548b25a62efee9cb0da7/packages/common/src/components/Button/index.tsx#L4

Comment (armujahid) [reply packages/app/src/app/pages/Dashboard/index.tsx]: Not sure whether I should do that or not.

Comment (armujahid) [reply packages/app/src/app/pages/Dashboard/index.tsx]: Fixed

File: packages/app/src/app/pages/Dashboard/index.js (hunk 1 above)
Comment (armujahid) [reply packages/app/src/app/pages/Dashboard/index.js]: Fixed

**Comment by armujahid** [comment]
All green again :)

**Pull request merged**